storm-core/src/jvm/backtype/storm/Config.java

```diff
@@ -1491,6 +1491,29 @@
     public static final String TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT = "topology.component.cpu.pcore.percent";
 
     /**
+     * The class name of the {@link backtype.storm.state.StateProvider} implementation. If not specified
+     * defaults to {@link backtype.storm.state.InMemoryKeyValueStateProvider}. This can be overridden
+     * at the component level.
+     */
+    @isString
+    public static final String TOPOLOGY_STATE_PROVIDER = "topology.state.provider";
+
+    /**
+     * The configuration specific to the {@link backtype.storm.state.StateProvider} implementation.
+     * This can be overridden at the component level.
+     */
+    @isString
+    public static final String TOPOLOGY_STATE_PROVIDER_CONFIG = "topology.state.provider.config";
```

Comment:
For now we expect the actual config string (for redis, its a json string with the host, port etc). it could be a file name as well, if the state provider implementation desires so.